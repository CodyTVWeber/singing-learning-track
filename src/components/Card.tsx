import React, { useState } from 'react';
import { colors, spacing, borderRadius, shadows, transitions, gradients, blurs } from '../theme/theme';

interface CardProps {
  children: React.ReactNode;
  padding?: keyof typeof spacing;
  onClick?: () => void;
  className?: string;
  style?: React.CSSProperties;
  variant?: 'default' | 'elevated' | 'outlined' | 'gradient' | 'glass';
  hoverEffect?: boolean;
  decorative?: boolean;
}

export const Card: React.FC<CardProps> = ({ 
  children, 
  padding = 'lg',
  onClick,
  className = '',
  style,
  variant = 'default',
  hoverEffect = true,
  decorative = false,
}) => {
  const [isHovered, setIsHovered] = useState(false);
  
  const baseStyles: React.CSSProperties = {
    borderRadius: borderRadius.xl,
    padding: spacing[padding],
    cursor: onClick ? 'pointer' : 'default',
    transition: transitions.smooth,
    position: 'relative',
    overflow: 'hidden',
  };
  
  const variantStyles: Record<string, React.CSSProperties> = {
    default: {
      backgroundColor: colors.surface,
      boxShadow: shadows.sm,
      border: `1px solid ${colors.gray100}`,
    },
    elevated: {
      backgroundColor: colors.surfaceElevated,
      boxShadow: shadows.lg,
      border: 'none',
    },
    outlined: {
      backgroundColor: 'transparent',
      boxShadow: 'none',
      border: `2px solid ${colors.gray200}`,
    },
    gradient: {
      background: gradients.soft,
      boxShadow: shadows.md,
      border: `1px solid rgba(255, 255, 255, 0.2)`,
    },
    glass: {
      backgroundColor: 'rgba(255, 255, 255, 0.8)',
      backdropFilter: `blur(${blurs.md})`,
      WebkitBackdropFilter: `blur(${blurs.md})`,
      boxShadow: shadows.md,
      border: `1px solid rgba(255, 255, 255, 0.3)`,
    },
  };
  
  const hoverStyles: React.CSSProperties = hoverEffect && onClick ? {
    transform: isHovered ? 'translateY(-4px) scale(1.02)' : 'translateY(0) scale(1)',
    boxShadow: isHovered ? shadows.xl : variantStyles[variant].boxShadow,
  } : {};

  return (
    <div
      className={className}
      style={{
        ...baseStyles,
        ...variantStyles[variant],
        ...hoverStyles,
        ...style,
      }}
      onClick={onClick}
      onMouseEnter={() => setIsHovered(true)}
      onMouseLeave={() => setIsHovered(false)}
    >
      {/* Decorative elements */}
      {decorative && variant === 'gradient' && (
        <>
          <div
            style={{
              position: 'absolute',
              top: '-50%',
              right: '-20%',
              width: '200px',
              height: '200px',
              background: gradients.ocean,
              borderRadius: '50%',
              filter: `blur(${blurs.xl})`,
              opacity: 0.3,
              transition: transitions.slow,
              transform: isHovered ? 'scale(1.2)' : 'scale(1)',
            }}
          />
          <div
            style={{
              position: 'absolute',
              bottom: '-50%',
              left: '-20%',
              width: '150px',
              height: '150px',
              background: gradients.sunset,
              borderRadius: '50%',
              filter: `blur(${blurs.xl})`,
              opacity: 0.3,
              transition: transitions.slow,
              transform: isHovered ? 'scale(1.2)' : 'scale(1)',
            }}
          />
        </>
      )}
      
      {/* Content */}
      <div style={{ position: 'relative', zIndex: 1 }}>
        {children}
      </div>
      
      {/* Hover glow effect */}
      {hoverEffect && isHovered && variant !== 'outlined' && (
        <div
          style={{
            position: 'absolute',
            inset: 0,
            background: 'rgba(74, 144, 226, 0.05)',
            pointerEvents: 'none',
            transition: transitions.fast,
          }}
        />
      )}
    </div>
  );
};