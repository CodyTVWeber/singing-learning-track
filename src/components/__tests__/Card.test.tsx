import React from 'react';
import { render } from '@testing-library/react';
import { Card } from '../Card';

describe('Card', () => {
  it('renders with default padding snapshot', () => {
    const { asFragment } = render(<Card>Content</Card>);
    expect(asFragment()).toMatchSnapshot();
  });
});

