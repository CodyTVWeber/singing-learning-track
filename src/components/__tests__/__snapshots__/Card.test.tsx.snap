// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > renders with default padding snapshot 1`] = `
<DocumentFragment>
  <div
    class=""
    style="border-radius: 20px; padding: 24px; cursor: default; transition: 400ms cubic-bezier(0.4, 0, 0.2, 1); position: relative; overflow: hidden; background-color: rgb(255, 255, 255); box-shadow: 0 2px 4px -1px rgba(0, 0, 0, 0.06), 0 1px 2px -1px rgba(0, 0, 0, 0.04); border: 1px solid rgb(245, 245, 240);"
  >
    <div
      style="position: relative; z-index: 1;"
    >
      Content
    </div>
  </div>
</DocumentFragment>
`;
